app/controllers/api/base_controller/renderer.rb

```diff
@@ -469,6 +469,19 @@ def action_validated?(resource, action_spec)
         end
         true
       end
+
+      def render_options(resource, data = {})
+        collection = collection_class(resource)
+        options = {
+          :metadata => {
+            :attributes         => collection.attribute_types,
+            :virtual_attributes => collection.virtual_attributes_to_define,
+            :relationships      => collection_config.subcollections(resource),
+            :data               => data
+          }
```

Comment:
attributes, virtual_attributes and relationships should be what's currently seen by the renderer.  see attr_physical? and attr_virtual?    

Also, these here are not arrays, but hashes.  Was thinking just array of names.  maybe leveraging collection attribute_names (both phys/virt), virtual_attribute_names and reflections.keys

I'm afraid exposing the attr types/schema here binds us with implementation (rails/our virtual stuff).